### PR TITLE
[master] Acme fixes

### PR DESCRIPTION
**Pull request by jeffclay**
#### Acme State
Added `dns_plugin_propogate_seconds` option to cert().
Added option to replace an existing staging/test certificate with a production cert.

#### Acme Module
Added `dns_plugin_propogate_seconds` option to cert(). This was previously in the doc string but not actually coded anywhere.
Added revoke() with optional reason.
Added delete().

### What issues does this PR fix or reference?
Fixes:
https://github.com/saltstack/salt/issues/63700
https://github.com/saltstack/salt/issues/64686

### Previous Behavior

1. `dns_plugin_propogate_seconds` only existed in doc strings.
2. After creating a certificate using `test_cert` flag then removing the flag, the result was nothing happening due to:
   1.  Acme state does check to see if cert exists and if it does it just checks if it needs renewed. No consideration for replacing a staging certificate. https://github.com/saltstack/salt/blob/75269c4ae22cce7cef1e681e2b1625368d6e18f5/salt/states/acme.py#L104 
   2. Acme module did not have capability to revoke or delete a certificate.

### New Behavior

1. `dns_plugin_propogate_seconds`  exists in Acme state and module
2. Acme state has new bool option `replace_staging` that will revoke and delete a staging certificate as long as "STAGING" is in the cert issuer dictionary and `test_cert` is no longer requested.
3. Acme module has new `revoke` capability which will delete a cert after revocation due to the `--non-interactive` cmd switch.
4. Acme module has new `delete` capability which deletes certificates. (No need to call this after revoke in sls state or Acme state, just seemed like a natural thing to add along with `revoke`.)

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [X] Docs
- [X] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [X] Tests written/updated

### Commits signed with GPG?
Yes

Please review [Salt's Contributing Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

**Comment by jeffclay** [comment]
This is a retry for [PR  #64685](https://github.com/saltstack/salt/pull/64685) since I mangled the re-base.

File: salt/modules/acme.py [hunk 1]
```diff
@@ -421,3 +426,87 @@ def needs_renewal(name, window=None):
         window = int(window)
 
     return _renew_by(name, window) <= datetime.datetime.today()
+
+
+def revoke(name, reason=None):
+    """
```
Comment (twangboy) [inline salt/modules/acme.py:432]: Needs a `.. version-added:: 3007` here

File: salt/modules/acme.py [hunk 2]
```diff
@@ -421,3 +426,87 @@ def needs_renewal(name, window=None):
...
+                }
+        else:
+            return {
+                "result": False,
+                "comment": f"Certificate {name} revocation failed with:\n{res['stderr']}",
+            }
+
+
+def delete(name):
+    """
```
Comment (twangboy) [inline salt/modules/acme.py:476]: Needs a `.. version-added:: 3007` here

**Review by twangboy — CHANGES_REQUESTED** [review]

File: salt/modules/acme.py (hunk 1 above)
Comment (jeffclay) [reply salt/modules/acme.py:432]: I'm not sure I follow.

Comment (jeffclay) [reply salt/modules/acme.py:432]: @twangboy can you elaborate on what you're referring to?

Comment (twangboy) [reply salt/modules/acme.py:432]: When you add a new module, or a new function to a module, or a new parameter to a function, you need to note the version that that new item was added. That is done using `.. version-added:: <the version>`.

For example: https://github.com/saltstack/salt/blob/master/salt/modules/file.py#L597

Comment (jeffclay) [reply salt/modules/acme.py:432]: ah, ok. i hadn't seen that.

**Review by twangboy — DISMISSED** [review]

**Review by twangboy — DISMISSED** [review]

**Comment by jeffclay** [comment]
@dwoz Do I need to do anything here?

**Review by twangboy — APPROVED** [review]

**Pull request closed without merging**